src/modalities/models/gpt2/gpt2_model.py

```diff
@@ -292,6 +299,8 @@ def parse_sharding_strategy_by_name(cls, name):
         config: RotaryTransformConfig | IdentityTransformConfig
 
     qkv_transforms: list[QueryKeyValueTransformConfig]
+    use_qk_norm: bool = False
+    qk_norm_config: Optional[LayerNormWrapperConfig] = None
```

Comment:
I would remove `use_qk_norm` and only apply qk norm if `qk_norm_config` is provided (i.e., ` is not None`)